packages/core/src/shared/clients/s3Client.ts

```diff
@@ -709,3 +741,89 @@ async function createSdkClient(regionCode: string): Promise<S3> {
 function clearInternalBucketCache(): void {
     ;(S3.prototype as any).bucketRegionCache = {}
 }
+
+/**
+ * A URI parser that can parse out information about an S3 URI
+ * Adapted from
+ * @see https://github.com/frantz/amazon-s3-uri/
+ */
+export function parseS3Uri(uri: string): [region: string, bucket: string, key: string] {
+    const endpointPattern = /^(.+\.)?s3[.-]([a-z0-9-]+)\./
+    const defaultRegion = 'us-east-1' // Default region for URI parsing, if region is not found
+    const parsedUri = url.parse(uri)
+    let bucket: string | undefined = undefined
+    let region: string = defaultRegion
+    let key: string | undefined = undefined
+
+    if (parsedUri.protocol === 's3:') {
+        bucket = parsedUri.host ?? undefined
+        if (!bucket) {
+            throw new Error(`Invalid S3 URI: no bucket: ${uri}`)
```

Comment:
Why do we want to lint the URI given? Can't we rely on S3 for that?

Reply:
S3 does not support GetObject using URI out of the box. The idea of this is to get the region, bucket, and key from the URI and use those parameters to perform GetObject.